omod/src/main/java/org/openmrs/module/pihcore/fragment/controller/dashboardwidgets/AppointmentsFragmentController.java

```diff
@@ -0,0 +1,76 @@
+package org.openmrs.module.pihcore.fragment.controller.dashboardwidgets;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.codehaus.jackson.JsonNode;
+import org.joda.time.DateTime;
+import org.openmrs.Patient;
+import org.openmrs.api.PatientService;
+import org.openmrs.module.appframework.domain.AppDescriptor;
+import org.openmrs.module.appointments.model.Appointment;
+import org.openmrs.module.appointments.model.AppointmentSearchRequest;
+import org.openmrs.module.appointments.model.AppointmentStatus;
+import org.openmrs.module.appointments.service.AppointmentsService;
+import org.openmrs.module.emrapi.patient.PatientDomainWrapper;
+import org.openmrs.ui.framework.annotation.FragmentParam;
+import org.openmrs.ui.framework.annotation.SpringBean;
+import org.openmrs.ui.framework.fragment.FragmentConfiguration;
+import org.openmrs.ui.framework.fragment.FragmentModel;
+
+public class AppointmentsFragmentController {
+    public void controller(@SpringBean("appointmentsService") AppointmentsService appointmentsService,
+                           @SpringBean("patientService") PatientService patientService,
+                           @FragmentParam("app") AppDescriptor app,
+                           FragmentConfiguration config,
+                           FragmentModel model) {
+
+        Object patientConfig = config.get("patient");
+        if (patientConfig == null ) {
+            patientConfig = config.get("patientId");
+        }
+        Patient patient = null;
+        if (patientConfig != null) {
+            if (patientConfig instanceof Patient) {
+                patient = (Patient) patientConfig;
+            }
+            else if (patientConfig instanceof PatientDomainWrapper) {
+                patient = ((PatientDomainWrapper) patientConfig).getPatient();
+            }
+            else if (patientConfig instanceof Integer) {
+                patient = patientService.getPatient((Integer)patientConfig);
+            }
+            else if (patientConfig instanceof String) {
+                patient = patientService.getPatientByUuid((String)patientConfig);
+            }
+        }
+        if (patient == null) {
+            throw new IllegalArgumentException("No patient found. Please pass patient into the configuration");
+        }
+
+        AppointmentSearchRequest appointmentSearchRequest = new AppointmentSearchRequest();
+        appointmentSearchRequest.setPatientUuid(patient.getUuid());
+        appointmentSearchRequest.setStartDate(new DateTime().withTimeAtStartOfDay().toDate());
```

Comment:
So we are explicitly not showing appointments scheduled for any day prior to the current day, but which still have a status indicating they have not been honored?

Reply:
Good points, was just going based on the ticket description, will flag you there...